pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -359,7 +359,15 @@ func (s *ServiceEntryStore) serviceEntryHandler(_, curr config.Config, event mod
 	fullPush := len(configsUpdated) > 0
 	// if not full push needed, at least one service unchanged
 	if !fullPush {
-		s.edsUpdate(serviceInstances, true)
+		if len(serviceInstances) == 0 {
+			var old []*model.ServiceInstance
+			for _, v := range oldInstances {
+				old = append(old, v...)
+			}
+			s.edsUpdate(old, true)
```

Comment:
Seems not right

Reply:
For se with label selector, len(serviceInstances) is always = 0 here

Reply:
And for endpoints case, i donot understand how you remove the old instances with  this